### PR TITLE
Create Entity specific hooks | Closes #2228

### LINKED ISSUE #2228: Create Model Entity Specific Context Hooks
see: https://github.com/eventespresso/event-espresso-core/issues/2149#issuecomment-574753125

> We should extract the use of `useContext` to a hook like `useDateContext` or `useTicketContext` to handover the importing of context to it to give us the data we need.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Creates `useDatetimeContext` and `useTicketContext` hooks
- Adds tests for the created hooks
- Updates the context consumer components to use the new hooks
- Closes #2228

### CODE REVIEW

File: assets/prototype/domain/eventEditor/datetimes/dateCard/EditDateButton.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import React, { useContext, CSSProperties } from 'react';
+import React, { CSSProperties } from 'react';
 import { Button } from '@blueprintjs/core/lib/esm';
-import { DateTimeContext } from '../../context/DateTimeProvider';
 import { EditItemButtonProps } from '../../types';
+import { useDatetimeContext } from './../../hooks';
 
 const EditDateButton: React.FC<EditItemButtonProps> = ({ position }): JSX.Element => {
-	const { editors, editorState } = useContext(DateTimeContext);
+	const { editors, editorState } = useDatetimeContext();
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/datetimes/dateCard/EditDateButton.tsx]: :+1:

File: assets/prototype/domain/eventEditor/hooks/useDatetimeContext.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { useContext } from 'react';
+
+import { DateTimeContext, DateTimeContextProps } from '../context/DateTimeProvider';
+
+const useDatetimeContext = (): DateTimeContextProps => {
+	const value = useContext<DateTimeContextProps>(DateTimeContext);
+	if (!value) {
+		throw new Error('useDatetimeContext must be used inside <DateTimeProvider> component');
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/hooks/useDatetimeContext.ts]: This looks like an dev-related error.
Maybe we might consider using `invariant`: https://github.com/zertosh/invariant

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/hooks/useDatetimeContext.ts]: I haven't used that before.

File: assets/prototype/domain/eventEditor/hooks/test/useDatetimeContext.test.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+
+import useDatetimeContext from '../useDatetimeContext';
+import DateTimeProvider from '../../context/DateTimeProvider';
+
+describe('useDatetimeContext', () => {
+	it('checks for empty id when used outside the context', () => {
+		const { result } = renderHook(() => useDatetimeContext());
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/hooks/test/useDatetimeContext.test.tsx]: thnx for adding tests 👍

**Pull request merged**